### ISSUE TITLE
parse error with SVG file

### ISSUE THREAD

**Issue by Sembiance** [lowlighter/libs#11]
Attempts to `parse` this file fails: [abydos.svg.txt](https://github.com/lowlighter/xml/files/7535563/abydos.svg.txt)

With error:
```
Error: Failed to parse XML, expected token: ^(?<selfclosed>\/?)> (got: http:/...)
    at consume (https://deno.land/x/xml@v1.0.2/parse.ts:200:11)
    at parseNode (https://deno.land/x/xml@v1.0.2/parse.ts:151:7)
    at parseNode (https://deno.land/x/xml@v1.0.2/parse.ts:162:21)
    at parseNode (https://deno.land/x/xml@v1.0.2/parse.ts:162:21)
    at parseNode (https://deno.land/x/xml@v1.0.2/parse.ts:162:21)
    at parseXML (https://deno.land/x/xml@v1.0.2/parse.ts:123:40)
    at parse (https://deno.land/x/xml@v1.0.2/parse.ts:58:24)
```

**Comment by Sembiance** [comment]
Hrm, looks like it's not valid XML, the first line is:
`<?xml version="1.0" encoding="UTF-8" standalone="no"?>`
That trailing question mark doesn't look quite right :)

The browser, inkscape, various node xml2json modules and svgo all accept the file ok, but they are probably just a bit more lax in parsing slightly invalid XML.

Since svgo accepts it and outputs sane XML that this parser here handles fine, I'll just feed all my SVG files through svgo first.

Closing this issue.

**Comment by lowlighter** [comment]
Actually you may be onto something.

The XML prolog does have a trailing question mark as per the spec.

Seems that the prolog parser is searching closing tag token (`>`) or selfclosing tag token (`/>`) instead:
https://github.com/lowlighter/xml/blob/4eab976ea519c4c3b58096476f65383c8907e01e/utils/parser.ts#L113-L116

It should use the condition `!this.#peek(tokens.prolog.end)` instead, but not sure how it passed the tests 😅

**Comment by lowlighter** [comment]
Seems fixed by #12 
https://dash.deno.com/playground/fat-owl-55
